core/src/com/unciv/ui/trade/DiplomacyScreen.kt

```diff
@@ -275,6 +275,20 @@ class DiplomacyScreen(val viewingCiv:CivilizationInfo):CameraStageBaseScreen() {
             }
         }
 
+        for (unique in viewingCiv.getMatchingUniques("Can spend Gold to annex or puppet a City-State that has been your ally for [] turns.")) {
```

Comment:
So if we have multiple uniques, multiple of the same buttons would appear? Should this just be an `if (viewingCiv.hasUnique("bla")) {`?

Especially since we don't seem to differentiate between the different uniques when setting the flags, so the multiple buttons would all turn enabled at exactly the same time, removing any possible difference between them

Reply:
On a related note, this function is getting quite large. Maybe it's time to split it up? Extracting this part would allow you to simply `if (!viewingCiv.hasUnique("bla")) return`, also reducing unnecessary indentation